services/frontend-service/src/ui/Warnings.tsx

```diff
@@ -0,0 +1,82 @@
+import * as React from 'react';
+import DeleteForeverIcon from '@material-ui/icons/DeleteForever';
+import DeleteOutlineIcon from '@material-ui/icons/DeleteOutline';
+import type { Environment, Release } from '../api/api';
+
+export enum DeployState {
+    Normal,
+    Undeploy,
+    Mixed,
+}
+
+export function getDeployState(name: string, environments: { [name: string]: Environment }): DeployState {
```

Comment:
Maybe we should add some unit tests for `Warnings.tsx`?

Reply:
We have unit tests for that already